### PR TITLE
E2E Test

### PR DESCRIPTION
**Pull request by daopunk**
The opendollar package is broken.

### CODE REVIEW

**Comment by daopunk** [comment]
@pi0neerpat 
Screenshot 2024-05-10 at 9 08 45 PM

**Review by pi0neerpat — COMMENTED** [review]
Depends on #2

File: src/contracts/ODSaviour.sol [hunk 1]
```diff
@@ -1,12 +1,168 @@
...
+    liquidatorReward = _newReward;
+    emit LiquidatorRewardSet(_newReward);
+  }
+
+  /**
+   * @dev
+   */
+  function setVaultStatus(uint256 _vaultId, bool _enabled) external onlyRole(SAVIOUR_TREASURY) {
+    _enabledVaults[_vaultId] = _enabled;
+
```
Comment (pi0neerpat) [inline src/contracts/ODSaviour.sol]: Add validation that ctype is enabled for this vault type

File: src/contracts/ODSaviour.sol [hunk 2]
```diff
@@ -1,12 +1,168 @@
...
+    _setupRole(SAVIOUR_TREASURY, saviourTreasury);
+    _setupRole(PROTOCOL, protocolGovernor);
+    _setupRole(PROTOCOL, liquidationEngine);
+  }
+
+  function isEnabled(uint256 _vaultId) external view returns (bool _enabled) {
+    _enabled = _enabledVaults[_vaultId];
+  }
+
+  function addCType(bytes32 _cType, address _tokenAddress) external onlyRole(SAVIOUR_TREASURY) {
```
Comment (pi0neerpat) [inline src/contracts/ODSaviour.sol]: Update to use `_cParams` and `ModifiablePerCollateral` pattern as in the LiquidationEngine

```
  /// @inheritdoc ModifiablePerCollateral
  function _initializeCollateralType(bytes32 _cType, bytes memory _collateralParams) internal override whenEnabled {
    (LiquidationEngineCollateralParams memory _liqEngineCParams) =
      abi.decode(_collateralParams, (LiquidationEngineCollateralParams));
    _setCollateralAuctionHouse(_cType, _liqEngineCParams.collateralAuctionHouse);
    _cParams[_cType] = _liqEngineCParams;
  }

  /// @inheritdoc Modifiable
  function _modifyParameters(bytes32 _param, bytes memory _data) internal override {
    if (_param == 'onAuctionSystemCoinLimit') _params.onAuctionSystemCoinLimit = _data.toUint256();
    else if (_param == 'accountingEngine') accountingEngine = IAccountingEngine(_data.toAddress());
    else if (_param == 'saviourGasLimit') _params.saviourGasLimit = _data.toUint256();
    else revert UnrecognizedParam();
  }
  ```

File: src/contracts/ODSaviour.sol [hunk 3]
```diff
@@ -1,12 +1,168 @@
...
+    collateralJoinFactory = ICollateralJoinFactory(_init.collateralJoinFactory.assertNonNull());
+    safeEngine = ISAFEEngine(address(safeManager.safeEngine()));
+    liquidatorReward = _init.liquidatorReward;
+
+    if (_init.saviourTokens.length != _init.cTypes.length) revert LengthMismatch();
+
+    // solhint-disable-next-line  defi-wonderland/non-state-vars-leading-underscore
+    for (uint256 i; i < _init.cTypes.length; i++) {
+      _saviourTokenAddresses[_init.cTypes[i]] = IERC20(_init.saviourTokens[i].assertNonNull());
+    }
```
Comment (pi0neerpat) [inline src/contracts/ODSaviour.sol]: - Use `_grantRole` instead of setup`
- Add `_grantRole(DEFAULT_ADMIN_ROLE, msg.sender);`

**Review by pi0neerpat — CHANGES_REQUESTED** [review]

File: src/contracts/ODSaviour.sol (hunk 2 above)
Comment (MrDeadCe11) [reply src/contracts/ODSaviour.sol]: this is done on saviour branch

File: src/contracts/ODSaviour.sol (hunk 1 above)
Comment (MrDeadCe11) [reply src/contracts/ODSaviour.sol]: done in saviour branch.

File: src/contracts/ODSaviour.sol (hunk 3 above)
Comment (MrDeadCe11) [reply src/contracts/ODSaviour.sol]: we are now using authorizable.

**Review by pi0neerpat — APPROVED** [review]

**Comment by daopunk** [comment]
closes #10 
@pi0neerpat retro-active issue tracking

**Pull request merged**